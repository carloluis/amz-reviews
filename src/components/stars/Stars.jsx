import React from 'react';
import PropTypes from 'prop-types';
import Star from './Star/Star';
import styles from './Stars.scss';

const options = Array(5)
    .fill(1)
    .map((x, i) => x + i);

class Stars extends React.PureComponent {
    render() {
        const { stars, toggleStar } = this.props;

        return (
            <div className={styles.container}>
                FILTER BY:
                <div className={styles.stars}>
                    {options.map(s => <Star key={s} number={s} onToggle={toggleStar} checked={stars.includes(s)} />)}
                </div>
            </div>
        );
    }
}

Stars.propTypes = {
    stars: PropTypes.arrayOf(PropTypes.number).isRequired,
    toggleStar: PropTypes.func.isRequired
};

export default Stars;
