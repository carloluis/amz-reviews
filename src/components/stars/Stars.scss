@import '../../styles/palette.scss';

.container {
    font-weight: 600;
    font-size: 14px;
    text-transform: uppercase;
    color: $grey;
}
