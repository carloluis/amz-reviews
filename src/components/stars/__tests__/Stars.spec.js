import React from 'react';
import renderer from 'react-test-renderer';
import Stars from '../Stars';

const PROPS = {
    stars: [1, 5],
    toggleStar: jest.fn()
};

describe('<Stars/>', () => {
    it('snapshot', () => {
        const result = renderer.create(<Stars {...PROPS} />).toJSON();
        expect(result).toMatchSnapshot();
    });
});
