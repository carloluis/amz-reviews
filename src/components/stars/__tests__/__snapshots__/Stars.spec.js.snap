// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stars/> snapshot 1`] = `
<div
  className="container"
>
  FILTER BY:
  <div
    className="stars"
  >
    <span
      className="container checked"
      onClick={[Function]}
    >
      <i
        className="square"
      />
      <span>
        1
         ★
      </span>
    </span>
    <span
      className="container"
      onClick={[Function]}
    >
      <i
        className="square"
      />
      <span>
        2
         ★
      </span>
    </span>
    <span
      className="container"
      onClick={[Function]}
    >
      <i
        className="square"
      />
      <span>
        3
         ★
      </span>
    </span>
    <span
      className="container"
      onClick={[Function]}
    >
      <i
        className="square"
      />
      <span>
        4
         ★
      </span>
    </span>
    <span
      className="container checked"
      onClick={[Function]}
    >
      <i
        className="square"
      />
      <span>
        5
         ★
      </span>
    </span>
  </div>
</div>
`;
